mnemonic/mnemonic.py

```diff
@@ -212,7 +211,7 @@ def check(self, mnemonic: str) -> bool:
         l = len(b)  # noqa: E741
         d = b[: l // 33 * 32]
         h = b[-l // 33 :]
-        nd = binascii.unhexlify(hex(int(d, 2))[2:].rstrip("L").zfill(l // 33 * 8))
+        nd = bytes.fromhex(hex(int(d, 2))[2:].zfill(l // 33 * 8))
```

Comment:
this is fishy
i mean, it was fishy in the preexisting code as well, but seeing `bytes.fromhex(hex(...))`...

instead: `nd = int(d, 2).to_bytes(l // 33 * 4, "big")`
also the `int(hexdigest(...), 16)` below could use a corresponding modification

here's a piece of code that does this nicer, but that's for another PR (similar changes would need to apply whenever `hexdigest` or `int(x, 16)` or `hex()` is seen
```python
# bip32:
# checksum_len = initial_entropy_len // 32
# total_len = initial_entropy_len + checksum_len
# hence total_len == checksum_len * 33
total_len = len(b)
checksum_len = l // 33
entropy_len = l - checksum_len
entropy_bits = b[: -checksum_len]
checksum_bits = b[-checksum_len:]
entropy = int(entropy_bits, 2).to_bytes(entropy_len // 8, "big")
checksum = int(checksum_bits, 2).to_bytes(checksum_len // 8, "big")
return hashlib.sha256(entropy).digest() == checksum
```

Reply:
I'm slightly confused whether you want the change in this PR or another, but changed the line you mentioned.